packages/components/src/components/BarcodeGenerator/BarcodeGenerator.tsx

```diff
@@ -24,7 +24,38 @@ export const BarcodeGenerator = (props: IBarcodeProps) => {
           .join('/');
         const array = JSONPointer.get(props.data, headPath);
         const values = array.map((item: any) => JSONPointer.get(item, `/${tailPath}`));
-        setValues(values);
+        const parsedValues = values.map((item: any) => {
+          const convertToGS1String = (obj: any) => {
+            if (typeof obj !== 'object' || obj === null) return '';
+
+            let result = '';
+
+            // Handle '01' (01) first if it exists
+            if ('01' in obj) {
+              let value = obj['01'];
+              if (!/^\d{8,14}$/.test(value)) throw new Error('GTIN must be 8-14 digits');
```

Comment:
The regex hardcoded in the `if` statement to check the `01` AI value here might be incorrect. I checked the GS1 codebase and noticed they use this regex `(\d{12,14}|\d{8})` to validate the `01` AI value. You may  double-check it to ensure it's correct.

Reply:
I checked and tested again, and both cases worked well but I changed to follow the GS1 regex